hms-api/src/repository/idea_comment-repository.ts

```diff
@@ -0,0 +1,120 @@
+import {
+  getClient,
+  safeTransformArray,
+  safeTransformSSMember,
+} from './dynamo-db';
+import Uuid from '../util/Uuid';
+import {
+  AttributeValue,
+  DeleteItemCommand,
+  GetItemCommand,
+  PutItemCommand,
+  QueryCommand,
+} from '@aws-sdk/client-dynamodb';
+import NotFoundError from '../error/NotFoundError';
+import Comment from './domain/IdeaComment';
+import InvalidStateError from '../error/InvalidStateError';
+import ideaComment from './domain/IdeaComment';
+
+const dynamoDbClient = getClient();
+
+export async function listComments(ideaId: Uuid): Promise<Comment[]> {
+  const output = await dynamoDbClient.send(
+    new QueryCommand({
+      TableName: process.env.COMMENT_TABLE,
+      IndexName: process.env.COMMENT_BY_IDEA_ID_INDEX,
+      KeyConditionExpression: 'ideaId = :iId',
+      ExpressionAttributeValues: {':iId': {S: ideaId}},
+    }),
+  );
+
+  const items = output.Items;
+  if (items) {
+    return items.map((item) => itemToComment(item));
+  }
+
+  throw new NotFoundError(`Comments for Idea with id: ${ideaId} not found`);
+}
+
+export async function putComment(comment: Comment): Promise<void> {
+  await dynamoDbClient.send(
+    new PutItemCommand({
+      TableName: process.env.COMMENT_TABLE,
+      Item: {
+        id: {S: comment.id},
+        userId: {S: comment.userId},
+        ideaId: {S: comment.ideaId},
+        text: {S: comment.text},
+        replyTo: {S: comment.replyTo},
```

Comment:
this will have to be changed since replyTo (or whatever the name will be) is nullable

Reply:
lol, you'd need to check the other spots where we have nullable values but in the end it needs to be something like this when the value is null

```
  replyTo: {NULL: true},
```

I think there is a utility function I wrote somewhere that does this, I know one exists but it might not be in it's own file.  If it isn't in it's own file it would be good to move it for re-use

Reply:
I'm looking at other bits of the code where I think we should be doing conversions like I mentioned here from one this to null.  I don't see us doing it however.  The only place I really see us doing it is when we have empty lists and we need to store them in DynamoDB.  It might be that this just works no problem 🤔 

... or it will be something we find as a bug later.

Either way I guess it's fine for now